src/Widgets/EntryPopover.vala

```diff
@@ -0,0 +1,134 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public abstract class Tasks.EntryPopover<T> : Gtk.EventBox {
+    public Gtk.Image image { get; set; }
```

Comment:
Thinking about it more, this should probably be a construction property/argument and it would be more convenient for it to just be an icon name than a Gtk.Image

Reply:
wasn't sure about the drawbacks if we just pass an icon name - but seems like I was overthinking this :)